packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -324,8 +327,8 @@ class XcodeProjectInterpreter {
       _executable,
       '-project',
       _fileSystem.path.absolute(projectPath),
-      '-target',
-      target,
+      if (scheme != null)
+        ...<String>['-scheme', scheme],
```

Comment:
Important change 2: use the scheme (flavor or default Runner) instead of the target.